pkg/kafka/config/config.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+const (
+	// SuffixBrokers is a suffix for the brokers flag
+	SuffixBrokers = ".brokers"
+	// SuffixTLS is a suffix for the tls flag
+	SuffixTLS = ".tls"
+	// SuffixCert is a suffix for the tls certificate path flag
+	SuffixCert = ".tls.cert"
+	// SuffixKey is a suffix for the tls key path flag
+	SuffixKey = ".tls.key"
+	// SuffixCA is a suffix for the tls ca path flag
+	SuffixCA = ".tls.ca"
+
+	// DefaultBroker is the default kafka broker
+	DefaultBroker = "127.0.0.1:9092"
+	// DefaultTLS is the default for TLS enabled
+	DefaultTLS = false
+	// DefaultCAPath is the default for the TLS CA path
+	DefaultCAPath = ""
+	// DefaultCertPath is the default for the TLS Cert path
+	DefaultCertPath = ""
+	// DefaultKeyPath is the default for the TLS key path
+	DefaultKeyPath = ""
+)
+
+// Configuration describes the shared configuration options for Producers or Consumers
+type Configuration struct {
+	Brokers []string
+	TLS     TLSConfiguration
+}
+
+// TLSConfiguration describes the configuration properties for TLS Connections to the Kafka Brokers
+type TLSConfiguration struct {
+	Enabled  bool
+	CertPath string
+	KeyPath  string
+	CaPath   string
+}
+
+// GetTLSConfig creates TLS Configuration
+func (tlsConfig TLSConfiguration) GetTLSConfig() (*tls.Config, error) {
+	ca, err := loadCA(tlsConfig.CaPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error reading ca")
+	}
+
+	cert, err := tls.LoadX509KeyPair(filepath.Clean(tlsConfig.KeyPath), filepath.Clean(tlsConfig.CertPath))
```

Comment:
I tested this PR and TLS support works for me expect that I had to switch the `kafka.producer.tls.cert` and `kafka.producer.tls.key` command line arguments otherwise it couldn't parse the certificate.
According to the documentation the parameter order should be (i.e. swapped):
```
func LoadX509KeyPair(certFile, keyFile string) (Certificate, error)
``` 
https://golang.org/pkg/crypto/tls/#LoadX509KeyPair

Reply:
thanks, i did miss that!